src/mpid/ch4/shm/gpu/gpu_pre.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#ifndef GPU_PRE_H_INCLUDED
+#define GPU_PRE_H_INCLUDED
+
+typedef struct {
+    int dummy;
+} MPIDI_GPU_Global_t;
+
+typedef struct {
+    int dummy;
+} MPIDI_GPU_global_t;
```

Comment:
We maintain two structures, one with a upper case and one with a lower case?  This is horrible.  :-)

Reply:
Yes. Again, these are not needed for the GPU IPC shmmod but I have them there to align to the XPMEM template and in case they are needed, for example when adding RMA support

Reply:
I am not sure what is the use of `MPIDI_xxx_Global_t` maybe @minsii can clarify?

Reply:
I have remove the upper case struct

Reply:
XPMEM does not use it. It simply followed the POSIX code...I think we can safely remove it for both XPMEM and GPU. We can address the XPMEM change after #4527